modules/analytics/databricks_workspace/workspace.tf

```
@@ -48,48 +48,15 @@ resource "azurerm_databricks_workspace" "ws" {
-        try(var.vnets[var.settings.custom_parameters.lz_key][var.settings.custom_parameters.vnet_key].subnets[var.settings.custom_parameters.public_subnet_key].id, null),
-        ),
-        null
-      )
-
-      private_subnet_name = try(coalesce(
-        try(var.settings.custom_parameters.private_subnet_name, null),
-        try(var.vnets[var.client_config.landingzone_key][var.settings.custom_parameters.vnet_key].subnets[var.settings.custom_parameters.private_subnet_key].name, null),
-        try(var.vnets[var.settings.custom_parameters.lz_key][var.settings.custom_parameters.vnet_key].subnets[var.settings.custom_parameters.private_subnet_key].name, null),
-        ),
-        null
-      )
+      public_subnet_network_security_group_association_id = can(var.settings.custom_parameters.public_subnet_network_security_group_association_id) || can(var.settings.custom_parameters.public_subnet_key) == false || can(var.settings.custom_parameters.public_subnet.key) == false ? try(var.settings.custom_parameters.public_subnet_network_security_group_association_id, null) : var.vnets[try(var.settings.custom_parameters.lz_key, var.client_config.landingzone_key)][var.settings.custom_parameters.vnet_key].subnets[var.settings.custom_parameters.public_subnet_key].id
+      private_subnet_name                                 = can(var.settings.custom_parameters.private_subnet_name) || can(var.settings.custom_parameters.private_subnet_key) == false || can(var.settings.custom_parameters.private_subnet.key) == false ? try(var.settings.custom_parameters.private_subnet_name, null) : var.vnets[try(var.settings.custom_parameters.lz_key, var.client_config.landingzone_key)][var.settings.custom_parameters.vnet_key].subnets[var.settings.custom_parameters.private_subnet_key].name
 
       #the NSG-association ID is the subnet-id, so it can be simplified:
-      private_subnet_network_security_group_association_id = try(coalesce(
-        try(var.settings.custom_parameters.private_subnet_network_security_group_association_id, null),
-        try(var.vnets[var.client_config.landingzone_key][var.settings.custom_parameters.vnet_key].subnets[var.settings.custom_parameters.public_subnet_key].id, null),
-        try(var.vnets[var.settings.custom_parameters.lz_key][var.settings.custom_parameters.vnet_key].subnets[var.settings.custom_parameters.public_subnet_key].id, null),
-        ),
-        null
-      )
-
+      private_subnet_network_security_group_association_id = can(var.settings.custom_parameters.private_subnet_network_security_group_association_id) || can(var.settings.custom_parameters.public_subnet_key) == false || can(var.settings.custom_parameters.public_subnet.key) == false ? try(var.settings.custom_parameters.private_subnet_network_security_group_association_id, null) : var.vnets[try(var.settings.custom_parameters.lz_key, var.client_config.landingzone_key)][var.settings.custom_parameters.vnet_key].subnets[var.settings.custom_parameters.public_subnet_key].id
```

```suggestion
      private_subnet_network_security_group_association_id = can(var.settings.custom_parameters.private_subnet_network_security_group_association_id) || can(var.settings.custom_parameters.public_subnet_key) == false ? try(var.settings.custom_parameters.private_subnet_network_security_group_association_id, null) : var.vnets[try(var.settings.custom_parameters.lz_key, var.client_config.landingzone_key)][var.settings.custom_parameters.vnet_key].subnets[var.settings.custom_parameters.public_subnet_key].id
```

Two problems here:

1. Removed this as not in the original and target implementation

```
|| can(var.settings.custom_parameters.public_subnet.key) == false
```

2. mixing private and public
private_subnet_network_security_group_association_id and using
var.settings.custom_parameters.public_subnet_key
Should it be var.settings.custom_parameters.private_subnet_key